src/spherely.pyi

```diff
@@ -234,42 +251,61 @@ boundary: _VFunc_Nin1_Nout1[Literal["boundary"], Geography, Geography]
 convex_hull: _VFunc_Nin1_Nout1[
     Literal["convex_hull"], PolygonGeography, PolygonGeography
 ]
-distance: _VFunc_Nin2optradius_Nout1[Literal["distance"], float, float]
-area: _VFunc_Nin1optradius_Nout1[Literal["area"], float, float]
-length: _VFunc_Nin1optradius_Nout1[Literal["length"], float, float]
-perimeter: _VFunc_Nin1optradius_Nout1[Literal["perimeter"], float, float]
+distance: _VFunc_Nin2optradius_Nout1[Literal["distance"], float, np.float64]
+area: _VFunc_Nin1optradius_Nout1[Literal["area"], float, np.float64]
+length: _VFunc_Nin1optradius_Nout1[Literal["length"], float, np.float64]
+perimeter: _VFunc_Nin1optradius_Nout1[Literal["perimeter"], float, np.float64]
 
 # io functions
 
-to_wkt: _VFunc_Nin1_Nout1[Literal["to_wkt"], str, object]
+to_wkt: _VFunc_Nin1optprecision_Nout1[Literal["to_wkt"], str, object]
 to_wkb: _VFunc_Nin1_Nout1[Literal["to_wkb"], bytes, object]
 
+@overload
+def from_wkt(
+    a: str,
+    oriented: bool = False,
+    planar: bool = False,
+    tessellate_tolerance: float = 100.0,
+) -> Geography: ...
+@overload
 def from_wkt(
-    a: Iterable[str],
+    a: list[str] | npt.NDArray[np.str_],
     oriented: bool = False,
     planar: bool = False,
     tessellate_tolerance: float = 100.0,
-) -> npt.NDArray[Any]: ...
+) -> T_NDArray_Geography: ...
+@overload
+def from_wkb(
+    a: bytes,
+    oriented: bool = False,
+    planar: bool = False,
+    tessellate_tolerance: float = 100.0,
+) -> Geography: ...
+@overload
 def from_wkb(
     a: Iterable[bytes],
     oriented: bool = False,
     planar: bool = False,
     tessellate_tolerance: float = 100.0,
-) -> npt.NDArray[Any]: ...
+) -> T_NDArray_Geography: ...
 
 class ArrowSchemaExportable(Protocol):
     def __arrow_c_schema__(self) -> object: ...
 
 class ArrowArrayExportable(Protocol):
     def __arrow_c_array__(
         self, requested_schema: object | None = None
-    ) -> Tuple[object, object]: ...
+    ) -> tuple[object, object]: ...
+
+class ArrowArrayHolder:
+    def __arrow_c_array__(self) -> tuple[object, object]: ...
```

Comment:
This looks like a bug then in the implementation. It should match the `ArrowArrayExportable` above

Reply:
OK, looking at the code, I wrote it to accept `*args, **kwargs` (I remember now that I struggled with making this more explicit keyword names),

So at least we should add that here to the signature as well. But ideally it is just the same as `ArrowArrayExportable` above (because that is the signature it should be in practice and that is useful for users to see)

Reply:
Ah yes I omitted `requested_schema` since the implementation returns an error when trying to pass anything other than the default value, but I'll update the signature since it still accepts the default value.